aws_analysis_tools/ec2_events/jira_listener.py

```diff
@@ -0,0 +1,150 @@
+    COMMENT_TEMPLATE = '{instance_name}\r\n\r\nPlease schedule Icinga downtime from {start_time} to {end_time}.'
+
+    def __init__(
+        self,
+        username,
+        password,
+        name=NAME,
+        logger=None,
+        stats=None,
+    ):
+        # Private variables, not to be used outside this module
+        self._name = NAME
+        self._logger = logger or get_logger(self._name)
+        self._stats = stats or get_stats(prefix=self._name)
+
+        self._headers = {
+            'Content-Type': 'application/json',
+            'Accept': 'application/json',
+        }
+        self._auth = (username, password)
+
+    def _request(self, **kwarg):
+        kwarg['headers'] = self._headers
+        kwarg['auth'] = self._auth
+        kwarg['url'] = self.URL_TEMPLATES['base'].format(url=kwarg['url'])
+
+        res = request(**kwarg)
+
+        if res.status_code > 299 or res.status_code < 200:
+            raise ValueError(
+                'Something went wrong. {status_code} {reason} was returned. Body: {body}'
+                .format(status_code=res.status_code, reason=res.reason, body=res.content)
+            )
+
+        return res.json()
+
+    def handle_event(self, instance, event):
+        issues = self._find_issues(instance)
+
+        for issue in issues:
+            self._comment_issue(issue, instance.tags['Name'], event)
+
+    def _find_issues(self, instance):
+        yesterday_str = (DateTime() - 1).Date()
+        jql_search = self.JQL_TEMPLATE.format(instance_id=instance.id, yesterday=yesterday_str)
+
+        res = self._request(
+            method='POST',
+            url=self.URL_TEMPLATES['search'],
+            json={
+                'jql': jql_search,
+                'fields': [],
+            },
+        )
+
+        issues = res['issues']
+        self._logger.debug('Found %s issues that matches the JQL search: %s', len(issues), jql_search)
+
+        return issues
+
+    def _comment_issue(self, issue, instance_name, event):
+        # GOTCHA: This is kinda dumb, but Jira does not return the comments when issues are searched
+        # Thus, the comments for each issue have to be pulled separately
+        comments_res = self._request(
+            method='GET',
+            url=self.URL_TEMPLATES['comment'].format(issue=issue['key'])
+        )
+        comments = comments_res['comments']
+
+        if len([c for c in comments if instance_name in c['body']]) < 1:
+            self._logger.debug('Determined issue %s needs a comment', issue['key'])
+
+            start_time = DateTime(event.not_before)
+            end_time = DateTime(event.not_after) if event.not_after is not None else DateTime(9999, 12, 31)
+            body = self.COMMENT_TEMPLATE.format(
+                instance_name=instance_name,
+                # GOTCHA: Change the time to PST for easier calculation
+                start_time=str(start_time.toZone('PST')),
```

Comment:
Why are we using PST? What happens when we switch to PDT in a few weeks; are the times in the comments off by an hour?